resources/autoscaling.rb

```diff
@@ -0,0 +1,169 @@
+property :return_info, [String, Array]
+property :should_decrement_desired_capacity,	[true, false], default: true
+property :asg_name,              String
+property :status_code,           String
+property :max_size,              Integer, default: 4
+
+# authentication
+property :aws_access_key,        String
+property :aws_secret_access_key, String
+property :aws_session_token,     String
+property :aws_assume_role_arn,   String
+property :aws_role_session_name, String
+property :region,                String, default: lazy { fallback_region }
+
+include AwsCookbook::Ec2 # needed for aws_region helper
+
+# allow use of the property names from the autoscaling cookbook
+alias_method :aws_access_key_id, :aws_access_key
+alias_method :aws_region, :region
+
+action :enter_standby do
```

Comment:
The parts of these actions that actually make changes should be wrapped in a converge_by block. This allows for a few things

1) Friendly update messages in logs
2) Proper resource convergence for notifications
3) Support for why-run mode where chef will actually skip anything in the block

Reply:
done